go/libraries/doltcore/sqle/index/dolt_index.go

```diff
@@ -873,7 +875,7 @@ func (di *doltIndex) HasContentHashedField() bool {
 }
 
 func (di *doltIndex) Order() sql.IndexOrder {
-	if di.HasContentHashedField() {
+	if di.indexSch == nil || di.HasContentHashedField() {
```

Comment:
It seems a little redundant to check `di.indexSch == nil` here, when it's included in `di.HasContentHashedField()`. Omitting that would make this logic a little simpler and a little more clear (especially if you comment/explain the nil check for `di.indexSch` in the `HasContentHashedField()` method.